src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java

```diff
@@ -105,6 +109,39 @@ void surrogatesWithRaw() throws Exception
         jp.close();
     }
 
+    @Test
+    void lastSegmentCharSplitSurrogateCharInTwoSegments() throws Exception
+    {
+        // segments split every 1000 chars.
+        // We need a string with length 1001 where the surrogate is
+        // at 1000 and 1001 positions
+        int count = 999;
+        char[] chars = new char[count];
+        java.util.Arrays.fill(chars, 'x');
+        String base = new String(chars);
+
+        final String VALUE = base + "\uD83E\uDEE1";
+
+        // alas, we have to pull the recycler directly here...
```

Comment:
Hmmh. Why are we using `BufferRecycler` at all here?

Reply:
Trying to simulate my own use case. I don't want a string at the end of this. I want a UTF-8 byte array.

Reply:
Ah, then just use `ByteArrayOutputStream`? But either way let's not use `BufferRecycler` unless it's required part of reproduction; keep it (mostly) internal thing.